crates/polars-ops/src/chunked_array/list/cum.rs

```diff
@@ -0,0 +1,55 @@
+use std::option::Option;
+use std::vec::Vec;
+
+use polars_core::prelude::*;
+
+pub fn list_cum_concat(a: &ListChunked) -> PolarsResult<ListChunked> {
+    let mut values: Vec<Option<Series>> = Vec::new();
+    let mut acc: Option<Series> = None;
+    for value in a {
+        match (&mut acc, &value) {
+            (Some(acc_inner), Some(v)) => {
+                acc_inner.append(v)?;
+                values.push(Some(acc_inner.clone()));
+            },
+            (None, Some(v)) => {
+                acc = Some(v.clone());
+                values.push(Some(v.clone()));
+            },
+            (_, None) => {
+                values.push(None);
+            },
+        }
+    }
+
+    let mut ca: ListChunked = values.into_iter().collect();
+    ca.rename(a.name());
+    Ok(ca)
+}
+
+pub fn list_cum_set_union(a: &ListChunked) -> PolarsResult<ListChunked> {
+    let mut values: Vec<Option<Series>> = Vec::new();
+    let mut acc: Option<Series> = None;
+    for value in a {
+        match (&mut acc, &value) {
+            (Some(acc_inner), Some(v)) => {
+                let mut new_acc = acc_inner.clone();
+                new_acc.append(v)?;
+                new_acc = new_acc.0.unique()?;
```

Comment:
I really don't think this is the way to go. This should be typed and keep a hashtable in `acc`. As it states now, it rebuilds a hash table on every iteration. This is not linear time, but quadratic.

Reply:
@andysham maybe polars plugins are a good fit for this?

https://github.com/pola-rs/pyo3-polars

Reply:
Was not aware of polars plugins - will definitely look into this, thanks!